File: Src2/Icm.Core/Functions/TotalOrder/DateTotalOrder.vb
Namespace Icm
    Public Class DateTotalOrder
        Inherits BaseTotalOrder(Of Date)

        Public Overrides Function Least() As Date
            Return Date.MinValue
        End Function

        Public Overrides Function Greatest() As Date
            Return Date.MaxValue
        End Function

        Public Overrides Function T2Long(ByVal t As Date) As Long
            Return t.Ticks
        End Function

        Public Overrides Function Long2T(ByVal d As Long) As Date
            Return New Date(d)
        End Function

    End Class
End Namespace

File: Src/Icm.Core/Functions/TotalOrder/DateTotalOrder.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
namespace Icm
{
	public class DateTotalOrder : BaseTotalOrder<System.DateTime>
	{

		public override System.DateTime Least()
		{
			return System.DateTime.MinValue;
		}

		public override System.DateTime Greatest()
		{
			return System.DateTime.MaxValue;
		}

		public override long T2Long(System.DateTime t)
		{
			return t.Ticks;
		}

		public override System.DateTime Long2T(long d)
		{
			return new System.DateTime(d);
		}

	}
}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
